yarn-project/noir-contracts/src/contracts/uniswap_contract/src/main.nr

```diff
@@ -49,6 +49,96 @@ contract Uniswap {
+            let selector = compute_selector(
+                "swap_public((Field),(Field),Field,(Field),Field,Field,Field,(Field),Field,Field,(Field),(Field),Field)"
+            );
+            let message_field = compute_message_hash([
+                context.msg_sender(),
+                context.this_address(),
+                selector,
+                sender.address,
+                input_asset_bridge.address,
+                input_amount,
+                output_asset_bridge.address,
+                nonce_for_transfer_approval,
+                uniswap_fee_tier,
+                minimum_output_amount,
+                recipient.address,
+                secret_hash_for_L1_to_l2_message,
+                deadline_for_L1_to_l2_message,
+                canceller_for_L1_to_L2_message.address,
+                caller_on_L1.address,
+                nonce_for_swap_approval,
+            ]);
+            // this also emits a nullifier for the message
+            assert_valid_public_message_for(&mut context,sender.address,message_field);
+        }
+
+        let input_asset = AztecAddress::new(TokenBridge::at(input_asset_bridge.address).token(context));
+
+        // Transfer funds to this contract
+        Token::at(input_asset.address).transfer_public(
+            context,
+            sender.address,
+            context.this_address(),
+            input_amount,
+            nonce_for_transfer_approval,
+        );
+
+        // Approve bridge to burn this contract's funds and exit to L1 Uniswap Portal
+        let _void = context.call_public_function(
+            context.this_address(), 
+            compute_selector("_approve_bridge_and_exit_input_asset_to_L1((Field),(Field),Field)"), 
+            [input_asset.address, input_asset_bridge.address, input_amount],
+        );
+
+        // Create swap message and send to Outbox for Uniswap Portal
+        // this ensures the integrity of what the user originally intends to do on L1.
+        let input_asset_bridge_portal_address = get_portal_address(input_asset_bridge.address);
+        let output_asset_bridge_portal_address = get_portal_address(output_asset_bridge.address);
+        assert(input_asset_bridge_portal_address != 0, "L1 portal address of input_asset's bridge is 0");
+        assert(output_asset_bridge_portal_address != 0, "L1 portal address of output_asset's bridge is 0");
+
+        let content_hash = compute_swap_public_content_hash(
+            input_asset_bridge_portal_address,
+            input_amount,
+            uniswap_fee_tier,
+            output_asset_bridge_portal_address,
+            minimum_output_amount,
+            recipient.address,
+            secret_hash_for_L1_to_l2_message,
+            deadline_for_L1_to_l2_message,
+            canceller_for_L1_to_L2_message.address,
+            caller_on_L1.address,
+        );
+        context.message_portal(content_hash);
+
+        1
```

Comment:
May aswell just return a bool if its being used for a check?

Reply:
This is probably a leftover from doing similar to what I did in some of the other contracts when bools was not supported, don't think we changed it other places as well 🤔

Reply:
Made #2615